docs/UserGuide.md

```diff
@@ -130,10 +153,71 @@ _Written by: Brian Wong Yun Long_
 
 ### Listing the Transactions: `list`
 
+
+Lists all or some transactions based on selection. If tag filters are used, the transactions retrieved from the records must
+match all the filter tags that have been specified in order to be recognized as a valid record.
+
+**Format:** `list [t/TYPE] [c/CATEGORY] [d/DATE] [m/MONTH] [y/YEAR] [p/PERIOD] [n/NUMBER]`
+
+| Field      | Description                                                                                                                        |
+|------------|------------------------------------------------------------------------------------------------------------------------------------|
+| `TYPE`     | The type of transaction. It should either be `expense` or `income`.                                                                |
+| `CATEGORY` | A category for the transaction. It is a one-word parameter flexibly defined by the user. No numeral, symbol or spacing is allowed. |
+| `DATE`     | The date when the transaction took place on. It must be in ddMMyyyy format, e.g. 29102022.                                         | 
+| `MONTH`    | The month which the transaction falls on. It is in numerical form, i.e. from 1 to 12, where 1 represents January.                  |                                                                                                 
+| `YEAR`     | The year which the transaction falls on. It must be in yyyy format and only year 1000 and onwards are accepted.                    | 
+| `PERIOD`   | The period which the transaction falls on. It should either be `weeks` or `months`                                                 |
+| `NUMBER`   | The last N number of weeks or months. It is a positive whole number that is from 1 to 100.                                         |
+
+**Important Information:**
+
+- In order to view transactions for a specific month via `m/MONTH` tag, it is mandatory to include a `y/YEAR` tag.
+- The `p/PERIOD` and `n/NUMBER` tags must be used as a pair for viewing transactions for the
+  last N number of weeks or months.
+- The `m/MONTH` and `y/YEAR` tags should not be used together with `p/PERIOD` and `n/NUMBER` tags.
```

Comment:
Great usage of table in the UG! I will change my part to follow your format too!